src/components/KTableData/KTableData.vue

```diff
@@ -0,0 +1,706 @@
+    default: false,
+  },
+  sortHandlerFunction: {
+    type: Function as PropType<(param: SortHandlerFunctionParam) => Record<string, any>[]>,
+    default: () => ({}),
+  },
+  sortable: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+   * A boolean to hide pagination when total table records number is less than or equal to page size
+   */
+  hidePaginationWhenOptional: {
+    type: Boolean,
+    default: false,
+  },
+  hideToolbar: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'cell-click', value: { data: any }): void
+  (e: 'row-click', value: { data: any }): void
+  (e: 'error-action-click'): void
+  (e: 'empty-state-action-click'): void
+  (e: 'update:table-preferences', preferences: TablePreferences): void
+  (e: 'sort', value: TableSortPayload): void
+  (e: 'state', value: TableStatePayload): void
+}>()
+
+const tableId = useUniqueId()
+
+const tableData = ref<Record<string, any>[]>([])
+const tableHeaders = computed((): TableDataHeader[] => props.sortable ? props.headers : props.headers.map((header) => ({ ...header, sortable: false })))
+const getEmptyStateButtonAppearance = computed((): ButtonAppearance => {
+  if (props.emptyStateButtonAppearance) {
+    return props.emptyStateButtonAppearance
+  }
+
+  return props.searchInput ? 'tertiary' : 'primary'
+})
+
+const total = ref<number>(0)
+const page = ref<number>(1)
+const pageSize = ref<number>(props.paginationAttributes?.initialPageSize || 15)
+const filterQuery = ref<string>('')
+const sortColumnKey = ref<string>('')
+const sortColumnOrder = ref<SortColumnOrder>('desc')
+const offset: Ref<string | null> = ref(null)
+const offsets: Ref<Array<any>> = ref([])
+const hasNextPage = ref<boolean>(true)
+const hasInitialized = ref<boolean>(false)
+
+const defaultFetcherProps = {
+  pageSize: pageSize.value,
+  page: 1,
+  query: '',
+  sortColumnKey: '',
+  sortColumnOrder: 'desc',
+  offset: null,
+}
+
+const tablePaginationAttributes = computed((): TablePaginationAttributes => ({
+  ...props.paginationAttributes,
+  totalCount: total.value,
+  initialPageSize: pageSize.value,
+  currentPage: page.value,
+}))
+
+const getHeaderSlots = computed((): TableColumnSlotName[] => {
+  if (!slots) {
+    return []
+  }
+
+  return Object.keys(slots).filter((slot) => slot.startsWith('column-')) as TableColumnSlotName[]
+})
+
+const getHeaderTooltipSlots = computed((): TableColumnTooltipSlotName[] => {
+  if (!slots) {
+    return []
+  }
+
+  return Object.keys(slots).filter((slot) => slot.startsWith('tooltip-')) as TableColumnTooltipSlotName[]
+})
+
+const getCellSlots = computed((): string[] => {
+  if (!slots) {
+    return []
+  }
+
+  return Object.keys(slots).filter((slot) => tableHeaders.value.some((header) => header.key === slot))
+})
+
+const isInitialFetch = ref<boolean>(true)
+const fetchData = async () => {
+  const searchInput = props.searchInput
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```

Comment:
question: why is the ts-ignore needed here? if absolutely needed, please add an inline comment next to the ignore explaining